File: Data Structures & Algorithms/non-cyclical-number/submission-1.py
class Solution:
    def isHappy(self, n: int) -> bool:
        slow, fast = n, self.sumSquareDigits(n)

        while slow != fast:
            slow = self.sumSquareDigits(slow)
            fast = self.sumSquareDigits(self.sumSquareDigits(fast))
        
        return True if fast == 1 else False
    
    def sumSquareDigits(self, n: int) -> int:
        res = 0
        while n > 0:
            res += (n % 10) * (n % 10)
            n = n // 10
        return res

File: Data Structures & Algorithms/pow-x-n/submission-1.cpp
class Solution {
public:
    double myPow(double x, int n) {
        if(n == 0) {
            return 1;
        }

        double temp = 1;
        if(n < 0) {
            x = 1 / x;
            if(n == INT_MIN) {
                n++;
                temp = x;
            }
            n *= -1;
        }

        double res = x;
        while (n > 1) {
            if(n % 2 == 1) {
                temp *= res;
            }
            res *= res;

            n /= 2;
        }
        
        return res * temp;
    }
};
